javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java

```diff
@@ -86,8 +136,13 @@ private static void addFileUrl(List<URL> result, File file) {
     }
   }
 
-  private static URLClassLoader getDelegate(ClassLoader parent, URL extensionUrl) {
-    return new ExtensionClassLoader(new URL[] {extensionUrl}, parent);
+  private static void extractFile(JarFile jarFile, JarEntry jarEntry, File outputFile)
```

Comment:
`outputDir` since we're extracting a jar?

Reply:
No, that parameter is actually a file

Reply:
Oh oops - so the output file is a JAR file itself. `extractFile` made me think the embedded JAR itself is getting extracted into its contents. I don't think `unpackFile` had the same impression for me so may be a clearer name for people like me

https://github.com/spring-projects/spring-boot/blob/v2.4.3/spring-boot-project/spring-boot-tools/spring-boot-loader/src/main/java/org/springframework/boot/loader/archive/JarFileArchive.java#L140

Reply:
We extract a file from jarEntry into an outputFile. Like "extract a file from a jar". How can this translate to a double extraction? :) But I can rename if you wish, no problem

Reply:
There's two "Jar" variables there :) My brain processed it as extracting the second one. Maybe it's silly, so don't mind either way